gemutils/src/common/soap/GEMSOAPToolBox.cc

```diff
@@ -759,3 +756,103 @@ std::string gem::utils::soap::GEMSOAPToolBox::getXSDType(xdata::Serializable con
+    
+  xdata::soap::Serializer serializer;
+  xdata::String tmpVal;
+  xoap::SOAPElement element = gem::utils::soap::GEMSOAPToolBox::extractSOAPCommandParameterElement(msg, parameterName);
+  try
+    {
+      serializer.import(&tmpVal, element.getDOM());
+    }
+  catch(xcept::Exception& err)
+    {
+      std::string const msg =
+        toolbox::toString("Failed to import SOAP command parameter 'xdaq:%s': '%s'.",
+                          parameterName.c_str(),
+                          err.message().c_str());
+      XCEPT_RAISE(gem::utils::exception::SOAPException, msg);
+      
+    }
+
+  return std::string(tmpVal);
+}
+
+
+int gem::utils::soap::GEMSOAPToolBox::extractSOAPCommandParameterInteger(xoap::MessageReference const& msg, std::string const& parameterName)
+{
+    log4cplus::Logger m_gemLogger(log4cplus::Logger::getInstance("GEMSOAPToolBoxLogger"));
+    
+  xdata::soap::Serializer serializer;
+  xdata::Integer tmpVal;
+  xoap::SOAPElement element = gem::utils::soap::GEMSOAPToolBox::extractSOAPCommandParameterElement(msg, parameterName);
+  try
+    {
+      serializer.import(&tmpVal, element.getDOM());
+    }
+  catch(xcept::Exception& err)
+    {
+      std::string const msg =
+        toolbox::toString("Failed to import SOAP command parameter 'xdaq:%s': '%s'.",
+                          parameterName.c_str(),
+                          err.message().c_str());
+     XCEPT_RAISE(gem::utils::exception::SOAPException, msg);
+      
+    }
+  return int(tmpVal);
+}
+
+bool gem::utils::soap::GEMSOAPToolBox::extractSOAPCommandParameterBoolean(xoap::MessageReference const& msg,  std::string const& parameterName)
+{
+    log4cplus::Logger m_gemLogger(log4cplus::Logger::getInstance("GEMSOAPToolBoxLogger"));
+    
+  xdata::soap::Serializer serializer;
+  xdata::Boolean tmpVal;
+  xoap::SOAPElement element = gem::utils::soap::GEMSOAPToolBox::extractSOAPCommandParameterElement(msg, parameterName);
+  try
+    {
+      serializer.import(&tmpVal, element.getDOM());
+    }
+  catch(xcept::Exception& err)
+    {
+      std::string const msg =
+        toolbox::toString("Failed to import SOAP command parameter 'xdaq:%s': '%s'.",
+                          parameterName.c_str(),
+                          err.message().c_str());
+      
+      XCEPT_RAISE(gem::utils::exception::SOAPException, msg);
+   
+    }
+  return bool(tmpVal);
+}
+
+
+
+xoap::SOAPElement gem::utils::soap::GEMSOAPToolBox::extractSOAPCommandParameterElement(xoap::MessageReference const& msg, std::string const& parameterName )
+{
+    log4cplus::Logger m_gemLogger(log4cplus::Logger::getInstance("GEMSOAPToolBoxLogger"));
+    
+    
+    for (auto li: msg->getSOAPPart().getEnvelope().getBody().getChildElements()) {
+        xoap::SOAPName name = (li).getElementName();
+        std::vector<xoap::SOAPElement> children = (li).getChildElements();
+        std::vector<xoap::SOAPElement>::iterator vi;
+       
+        for (vi = children.begin(); vi != children.end(); ++vi)
```

Comment:
range-based loops have to be used

Reply:
I believe that the range-based for-loops modernization can done in all the code at once.

Reply:
Yes, I agree. The same applies to code migration towards JS or html templates.